metricbeat/module/redis/redis.go

```diff
@@ -76,6 +76,97 @@ func FetchSlowLogLength(c rd.Conn) (int64, error) {
 	return count, nil
 }
 
+// FetchKeyInfo collects info about a key
+func FetchKeyInfo(c rd.Conn, key string) (map[string]interface{}, error) {
+	keyType, err := rd.String(c.Do("TYPE", key))
+	if err != nil {
+		return nil, err
+	}
+	if keyType == "none" {
```

Comment:
We could make this a constant.

Reply:
Umm, also for the rest of types? Maybe the redis library already has already constants, I'll take a look.

Reply:
Using constants for types names now.